### ISSUE TITLE
[swarm] Manager node not able to list hosts -- rpc error: code = 4

### ISSUE THREAD

**Issue by rogaha** [docker/docker#25009]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Wed Jul 13 03:28:51 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.0-rc4
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   e4a0dbc
 Built:        Wed Jul 13 03:39:43 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.0-rc4
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: active
 NodeID: b0oprcm5gvg3nuuhibo9vibut
 IsManager: Yes
 Managers: 0
 Nodes: 0
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-boot2docker
Operating System: Boot2Docker 1.12.0-rc4 (TCL 7.2); HEAD : cbe6927 - Wed Jul 13 14:19:29 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995.9 MiB
Name: master2
ID: AJ2J:HTRF:6V2Q:KG3P:OZY5:WXXT:DGTJ:VJE5:DUVX:LLAY:LQDU:MEGS
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 30
 Goroutines: 90
 System Time: 2016-07-25T14:48:26.749584994Z
 EventsListeners: 0
Username: rogaha
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. From master2 node:

```
rogaha@Robertos-MacBook-Pro:~$ docker swarm join --manager --secret 9w16zuvfiplmqqt8ay0xlr57a \                                                                            
    --ca-hash sha256:xxxx \
    x.x.x.x:2377
Error response from daemon: Your node is in the process of joining the cluster but needs to be accepted by existing cluster member.
To accept this node into cluster run "docker node accept b0oprcm5gvg3nuuhibo9vibut" in an existing cluster manager. Use "docker info" command to see the current Swarm status of your node.
```

2. From the master node:

```
rogaha@Robertos-MacBook-Pro:~$ docker node ls                                                                                                                                7:47:08
ID                           HOSTNAME  MEMBERSHIP  STATUS   AVAILABILITY  MANAGER STATUS
2qn0dn86rl5udua48yln3l1ha    node2     Accepted    Ready    Active
48q6f972f5r28ujo86gt9p6kk    node1     Accepted    Ready    Active
4rp6nkie52ylfvvm2gxbx6dkv    node3     Accepted    Ready    Active
b0oprcm5gvg3nuuhibo9vibut              Pending     Unknown  Active
df6gzrjv1n81e13jgr5bxp2z8 *  master1   Accepted    Ready    Active        Leader
rogaha@Robertos-MacBook-Pro:~$ docker node accept b0oprcm5gvg3nuuhibo9vibut                                                                                                  
Node b0oprcm5gvg3nuuhibo9vibut accepted in the swarm.
rogaha@Robertos-MacBook-Pro:~$ docker node ls                                                                                                                                
ID                           HOSTNAME  MEMBERSHIP  STATUS  AVAILABILITY  MANAGER STATUS
2qn0dn86rl5udua48yln3l1ha    node2     Accepted    Ready   Active
48q6f972f5r28ujo86gt9p6kk    node1     Accepted    Ready   Active
4rp6nkie52ylfvvm2gxbx6dkv    node3     Accepted    Ready   Active
b0oprcm5gvg3nuuhibo9vibut    master2   Accepted    Ready   Active        Reachable
df6gzrjv1n81e13jgr5bxp2z8 *  master1   Accepted    Ready   Active        Leader
```

**Describe the results you received:**

From master2 node:
```
rogaha@Robertos-MacBook-Pro:~$ docker node ls                                                                                                                             
Error response from daemon: rpc error: code = 4 desc = context deadline exceeded
```

**Describe the results you expected:**

From master2 node:
```
rogaha@Robertos-MacBook-Pro:~$ docker node ls                                                                                                                                
ID                           HOSTNAME  MEMBERSHIP  STATUS  AVAILABILITY  MANAGER STATUS
2qn0dn86rl5udua48yln3l1ha    node2     Accepted    Ready   Active
48q6f972f5r28ujo86gt9p6kk    node1     Accepted    Ready   Active
4rp6nkie52ylfvvm2gxbx6dkv    node3     Accepted    Ready   Active
b0oprcm5gvg3nuuhibo9vibut    master2   Accepted    Ready   Active        Reachable
df6gzrjv1n81e13jgr5bxp2z8 *  master1   Accepted    Ready   Active        Leader
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by thaJeztah** [comment]
Probably relates to https://github.com/docker/docker/issues/24941 @abronan ?

**Comment by rogaha** [comment]
@thaJeztah on the docs we don't have any mentions to the

**Comment by thaJeztah** [comment]
@rogaha the procedure for joining a node has changed on master; https://github.com/docker/docker/pull/24823, I suspect the docs are currently somewhere in between RC4 and Master 😞

**Comment by rogaha** [comment]
got it, thanks for the heads up @thaJeztah!

**Comment by rogaha** [comment]
This can be closed, it's no longer and issue with `docker 1.12`.
